src/v/cloud_storage_clients/abs_client.cc

```diff
@@ -273,17 +273,33 @@ abs_request_creator::make_get_account_info_request() {
 abs_client::abs_client(
   const abs_configuration& conf,
   ss::lw_shared_ptr<const cloud_roles::apply_credentials> apply_credentials)
-  : _requestor(conf, std::move(apply_credentials))
+  : _data_lake_v2_client_config(
+    conf.is_hns_enabled ? std::make_optional(conf.make_adls_configuration())
```

Comment:
nit: it'd be great if you could sprinkle some of the CL or commit message's detail into a comment either here or around the member declaration. The connection between ADLS and HNS isn't obvious with fresh eyes

Reply:
Sure, can do.

Reply:
Added a comment at the member declaration site